sdk/keyvault/keyvault-common/test/internal/challengeAuthenticationCallbacks.spec.ts

```diff
@@ -89,6 +89,45 @@ describe("Challenge based authentication tests", function () {
 
       expect(options.request.headers.get("authorization")).toBeUndefined();
     });
+
+    it("Handles insufficient claims", async () => {
+      const options: AuthorizeRequestOptions = {
+        getAccessToken: () => {
+          return Promise.resolve({ token: "access_token", expiresOnTimestamp: 1000 });
+        },
+        request: createPipelineRequest({
+          url: "https://foo.bar",
+          headers: createHttpHeaders({}),
+        }),
+        scopes: [],
+      };
+
+      let called = false;
+
+      await challengeCallbacks.authorizeRequestOnChallenge!({
+        getAccessToken: (scopes, getAccessTokenOptions) => {
+          expect(scopes).to.deep.equal(["https://vault.azure.net/.default"]);
+          expect(getAccessTokenOptions.claims).to.equal(`{"claim":"fooo"}`);
+          expect(getAccessTokenOptions.enableCae).toBeTruthy();
+          called = true;
+          return Promise.resolve({ token: "successful_token", expiresOnTimestamp: 999999999 });
+        },
+        request,
+        response: {
+          headers: createHttpHeaders({
+            "WWW-Authenticate": `Bearer resource="https://vault.azure.net" error="insufficient_claims" claims="eyJjbGFpbSI6ImZvb28ifQ=="`,
```

Comment:
A real value looks like `Bearer realm="", authorization_uri="https://login.microsoftonline.com/common/oauth2/authorize", error="insufficient_claims", claims="eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ=="`. So, parameters are comma-delimited. Also, probably a good idea to test different parameter orderings because that isn't specified

Reply:
Oops. We do already test the WWW-authenticate parsing logic later on in this file, but I will update this case to follow the correct format.